src/nni_manager/training_service/local/gpuScheduler.ts

```diff
@@ -321,51 +83,18 @@ class GPUScheduler {
         return [];
     }
 
-    public stop(): void {
+    public async stop() {
         this.stopping = true;
+        const pid: string = await fs.promises.readFile(path.join(this.gpuMetricCollectorScriptFolder, 'pid'), 'utf8');
+        await cpp.exec(`pkill -P ${pid}`);
+        await cpp.exec(`rm -rf ${this.gpuMetricCollectorScriptFolder}`);
     }
 
-
-    private generateEmbededGPUSummary(data: nodeNvidiaSmi.GPUInfo) : GPUInfo[] {
-        let gpuInfos : GPUInfo[] = [];
-        const gpuNumber : number = parseInt(data.nvidia_smi_log.attached_gpus, 10);
-
-        assert(gpuNumber > 0);
-        if(gpuNumber == 1) {
-            const embededGPUSummary = <nodeNvidiaSmi.EmbededGPUSummary>data.nvidia_smi_log.gpu;
-            gpuInfos.push(this.convertGPUSummaryToInfo(embededGPUSummary));
-        } else {
-            const embededGPUSummaryArray = <nodeNvidiaSmi.EmbededGPUSummary[]>data.nvidia_smi_log.gpu;
-            gpuInfos = embededGPUSummaryArray.map(embededGPUSummary => this.convertGPUSummaryToInfo(embededGPUSummary));
+    private async updateGPUSummary() {
+        const cmdresult = await cpp.exec(`tail -n 1 ${path.join(this.gpuMetricCollectorScriptFolder, 'gpu_metrics')}`);
+        if(cmdresult && cmdresult.stdout) {
+            this.gpuSummary = <GPUSummary>JSON.parse(cmdresult.stdout);
```

Comment:
is there error log, if gpu metric is not correctly read?

Reply:
added log information.